### PR TITLE
Fixes #369: added support to deploy preview to surge

### LINKED ISSUE #369: deploy preview to surge
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

preview is generated and kept in heroku. after heroku redeployment preview is getting expired.
### Expected behaviour

generate the preview and deploy the preview to surge

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

The preview is getting expired because of heroku clear the persistent storage. so, I'm deploying to surge to show the preview.
## Related Issue

Close #369 
## Types of changes

- [ ] Bug fix 
- [ ] New feature

## Checklist:

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] There is a corresponding issue for this pull request.
- [ ] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [ ] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
if the surge login credentials are not provided, display logs on a sub route as before

**Comment by sch00lb0y** [comment]
@imujjwal96 I have updated. please check.

File: backend/deploy.js [hunk 1]
```diff
@@ -84,3 +84,21 @@ exports.deployHeroku = function (socket, data) {
...
+
+exports.deploySurge = function(data, surgeLogin, surgeToken, callback) {
+  var args = [
+    "-l", surgeLogin,
+    "-t", surgeToken,
+    "-e", data.email,
+    "-u", data.uniqueId
+  ];
+
+  var process = spawn('./surge_deploy.sh', args);
```
Comment (imujjwal96) [inline backend/deploy.js]: Change this to `spawnedProcess` like you've done in `generator.js`

File: util/logger.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+const mailer = require('../backend/mailer')
+BuildLog = require('../model/buildlog');
+
+/**
+ * Store the generated logs
+ * @param {Object} data : Data for storing logs
+ * @param {String} data.uniqueId : Unique id of the build
+ * @param {String} data.gitUrl : GIT URL of the repository
+ * @param {String} data.surgeDeploy : Flag to check whether it is surge deploy or not
+ * @param {string} data.surgeSuccessFlag : Flag to check whether it is deployed to surge or not
```
Comment (imujjwal96) [inline util/logger.js]: small s - capital s

File: util/logger.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * @param {String} data.email : email of the user
+ * @param {Integer} data.exitCode : Exit code of the process
+ */
+exports.storeLogs = function(data, callback) {
+  if (data.exitCode === 0) {
```
Comment (imujjwal96) [inline util/logger.js]:
```
if (date.exitCode !== 0) {
  return;
}

if (..
....
....
```
Try to use less number of if's. Reduce the cyclomatic complexity

**Comment by imujjwal96** [comment]
Where is the deployment? Update the description. Try to follow the PR template as much  as possible

File: util/logger.js (hunk 2 above)
Comment (sch00lb0y) [inline util/logger.js]: are you asking me to change `surgeSuccessFlag` to `SurgeSuccessFlag` ?

File: util/logger.js (hunk 3 above)
Comment (sch00lb0y) [inline util/logger.js]: @imujjwal96 I'm not able to minimize the if case. could you suggest me a way to minimize this ?

File: util/logger.js (hunk 2 above)
Comment (imujjwal96) [inline util/logger.js]: `String`

File: util/logger.js (hunk 3 above)
Comment (imujjwal96) [inline util/logger.js]: Oh! my mistake. I meant to say that you should try to  avoid nesting if. See the code below. I think it is correct. 
```
if (data.exitCode !== 0 && callback !== undefined) {
  BuildLog.storeGenerateLogs(reponame,
    'temp/' + email + '/generate_' + uniqueId + '.txt', function (error) {
      callback(error, data)
    });
  return callback({
    message: `Process exited with code : ${data.exitCode}`
  });
}

if (callback !== undefined) {
  BuildLog.storeGenerateLogs(miscellaneous.getRepositoryFullName(data.gitUrl),
    `temp/${data.email}/generate_${data.uniqueId}.txt`, function(error) {
      callback(error, data);
  });
  return;
}
if (!data.surgeDeploy ) {
  return mailer.sendEmail(data);
}

if (data.surgeSuccessFlag === true) {
  mailer.sendEmail(data);
}

```

**Comment by sch00lb0y** [comment]
@imujjwal96 updated. deployment: http://demo08.herokuapp.com/

**Comment by imujjwal96** [comment]
@sch00lb0y Surge works nicely! Can you remove surge credentials from environment vars?

**Comment by sch00lb0y** [comment]
@imujjwal96 I have removed please check now

**Pull request merged**